thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/timeseries/TimeSeriesResponseParser.java

```diff
@@ -1,381 +1,306 @@
-    return rowBuilderCache.buildAllRows(timeSeriesRequest.getStart(), timeSeriesRequest.getEnd());
-  }
-
-  /** Computes the OTHER category and adds it into the appropriate rows. */
-  private RowBuilderCache computeOtherRows(RowBuilderCache rowBuilderCache,
-      Map<MetricFunction, Double> metricOnlyData,
-      DoubleKeyMap<MetricFunction, String, Double> metricSumFromGroupByResponse) {
-    for (Entry<MetricFunction, Map<String, Double>> metricEntry : metricSumFromGroupByResponse
-        .toMap().entrySet()) {
-      MetricFunction metricFunction = metricEntry.getKey();
-      double total = metricOnlyData.get(metricFunction);
-      for (Entry<String, Double> dimensionEntry : metricEntry.getValue().entrySet()) {
-        String dimensionName = dimensionEntry.getKey();
-        double dimensionSum = dimensionEntry.getValue();
-        double difference = total - dimensionSum;
-        // only add OTHER group if the sum is different from the total.
-        if (difference > 0) {
-          Builder builder = rowBuilderCache.get(dimensionName, OTHER);
-          builder.addMetric(metricFunction.getMetricName(), difference);
-        }
-      }
+    boolean hasGroupByTime = false;
+    if (aggTimeGranularity != null) {
+      hasGroupByTime = true;
     }
-    return rowBuilderCache;
 
-  }
+    metricFunctions = response.getMetricFunctions();
+    numMetrics = metricFunctions.size();
+    numTimeBuckets = ranges.size();
+    rows = new ArrayList<>();
 
-  /**
-   * Extracts all dimension values for the first grouped dimension according to each
-   * request-response pairing. Return type is of form:
-   * MetricFunction->Dimension->DimensionValue->MetricValue
-   * This return format may change once multi-dimension grouping is supported.
-   * Behavior is undefined when multiple responses contain the same MetricFunction.
-   * This method ignores any ThirdEyeResponses for which the request reference is
-   * {@value TimeSeriesThirdEyeRequestGenerator#ALL}
-   */
-  Map<MetricFunction, Map<String, Map<String, Double>>> processGroupByDimensionResponse(
-      Map<ThirdEyeRequest, ThirdEyeResponse> queryResultMap) {
-    TripleKeyMap<MetricFunction, String, String, Double> metricGroupByDimensionData =
-        new TripleKeyMap<>(false, true, true);
-
-    for (Entry<ThirdEyeRequest, ThirdEyeResponse> entry : queryResultMap.entrySet()) {
-      ThirdEyeRequest thirdEyeRequest = entry.getKey();
-      ThirdEyeResponse thirdEyeResponse = entry.getValue();
-      if (CollectionUtils.isNotEmpty(thirdEyeRequest.getGroupBy())) {
-        extractMetricGroupByDimensionResponse(metricGroupByDimensionData, thirdEyeRequest,
-            thirdEyeResponse);
+    if (hasGroupByTime) {
+      if (hasGroupByDimensions) { // contributor view
+        parseGroupByTimeDimensionResponse();
+      } else { // tabular view
+        parseGroupByTimeResponse();
+      }
+    } else {
```

Comment:
Do we need the else block here? Timeseries won't really be without atleast group by time.
Also the comments here still say contributor/heatmap/tabular